python/marvin/tools/rss.py

```diff
@@ -117,65 +131,63 @@ def download(self):
 
         return super(RSS, self).download('mangarss', ifu=ifu, drpver=self._drpver, plate=plate)
 
-    def _load_rss_from_file(self):
-        """Initialises the RSS object from a file."""
+    def load_all(self):
+        """Loads all the `.RSSFiber` associated to this `.RSS` instance."""
 
-        try:
-            self.data = fits.open(self.filename)
-            self.mangaid = self.data[0].header['MANGAID'].strip()
-            self.plateifu = '{0}-{1}'.format(
-                self.data[0].header['PLATEID'], self.data[0].header['IFUDSGN'])
-        except Exception as ee:
-            raise MarvinError('Could not initialize via filename: {0}'.format(ee))
+        for rssfiber in self:
+            if not rssfiber.loaded:
+                rssfiber.load()
 
-        # Checks and populates release.
-        file_drpver = self.data[0].header['VERSDRP3']
-        file_drpver = 'v1_5_1' if file_drpver == 'v1_5_0' else file_drpver
+    def _load_rss_from_file(self, data=None):
+        """Initialises the RSS object from a file."""
 
-        file_ver = marvin.config.lookUpRelease(file_drpver)
-        assert file_ver is not None, 'cannot find file version.'
+        if data is not None:
+            assert isinstance(data, fits.HDUList), 'data is not an HDUList object'
+        else:
+            try:
+                self.data = fits.open(self.filename)
+            except (IOError, OSError) as err:
+                raise OSError('filename {0} cannot be found: {1}'.format(self.filename, err))
 
-        if file_ver != self._release:
-            warnings.warn('mismatch between file version={0} and object release={1}. '
-                          'Setting object release to {0}'.format(file_ver, self._release),
-                          MarvinUserWarning)
-            self._release = file_ver
+        self.header = self.data[1].header
+        self.wcs = astropy.wcs.WCS(self.header)
+        self.wcs = self.wcs.dropaxis(1)  # The header creates an empty axis for the exposures.
 
-        self._drpver, self._dapver = marvin.config.lookUpVersions(release=self._release)
+        # Confirm that this is a RSS file
+        assert 'XPOS' in self.data and self.header['CTYPE1'] == 'WAVE-LOG', \
+            'invalid file type. It does not appear to be a LOGRSS.'
 
-    def _load_rss_from_db(self):
-        """Initialises the RSS object from the DB."""
+        self._wavelength = self.data['WAVE'].data
+        self._shape = None
+        self._nfibers = self.data['FLUX'].shape[0]
 
-        import sqlalchemy
+        self.obsinfo = astropy.table.Table(self.data['OBSINFO'].data)
 
-        mdb = marvin.marvindb
+        Cube._do_file_checks(self)
 
-        if not mdb.isdbconnected:
-            raise MarvinError('No db connected')
+    def _load_rss_from_db(self, data=None):
```

Comment:
I'm sort of ok with this for remote calls but if I have a local db, I feel like it should do use it.  A local file versus db access of a 127 fiber RSS file is 12 seconds vs 15 seconds (in the old RSS code).  That's not a huge difference.  And that's without indices on the relevant columns you're grabbing?  Moving away from a database, in general, doesn't sit well with me, but that's a larger conversation.

Reply:
The problem here is that the DB doesn't allow you to create an `RSS` object with all the information that you could get from the file. And some of that information missing in the DB is important, such as the header or the obsinfo table. If we add those to the DB I'm happy to rethink this, but I feel that shouldn't be a priority right now.

Reply:
Actually, it turns out the `OBSINFO` is stored in the database.  In the `mangadatadb.obsinfo` table.  It's linked to the `mangadatadb.cube` table.

Reply:
I hadn't seen that. Then it's probably not that difficult to get all the information in the DB. I don't think we need an RSS table, we can use the cube one for linking, but maybe it's cleaner to have it. I think using the DB to retrieve these spectra will be efficient. Our bottleneck for the Cube is when we want to reconstruct the datacube, since that means retrieving a large number of spectra (1156 for the smallest cubes), but the number of spectra associated with an RSS is small (171 flux spectra for a 19-fibre IFU).